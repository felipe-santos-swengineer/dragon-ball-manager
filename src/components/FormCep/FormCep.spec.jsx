import React from 'react'
import { render, fireEvent, act } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import FormCep from "./FormCep"

//fazendo o pc entender comandos fora do js, como o fetch
global.fetch = jest.fn().mockImplementation(() => 
    Promise.resolve({
        json: () => ({
            cep: "03047000",
            logradouro: "Rua escorpiÃ£o",
            bairro: "Estrelas"
        })
    })
)



it("should render formCep", async() => {
    const { debug, getByPlaceholderText, container } = render(<FormCep />)
    //debug()
    //testando a entrada do campo placeholder do tipo input
    const cepInput = getByPlaceholderText('CEP').closest('input')
    //auto-preenchendo o campo
    fireEvent.change(cepInput, {target: {value:"03047000"}})

    await act(() => global.fetch)
    expect(global.fetch).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
    debug()
})
