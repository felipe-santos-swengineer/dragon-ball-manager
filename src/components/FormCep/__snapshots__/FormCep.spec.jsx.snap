// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render formCep 1`] = `
<div>
  <form
    class="css-1ldrcmr"
  >
    <div
      class="css-1viozcy"
    >
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="cep"
        >
          Cep
        </label>
        <input
          class="css-4y926w"
          id="cep"
          name="cep"
          placeholder="CEP"
          type="text"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="bairro"
        >
          Bairro
        </label>
        <input
          class="css-4y926w"
          id="bairro"
          name="bairro"
          placeholder="bairro"
          type="text"
          value="Estrelas"
        />
      </div>
      <div
        class="css-1e8zq2i"
      >
        <label
          class="css-knfhfe"
          for="logradouro"
        >
          Logradouro
        </label>
        <input
          class="css-4y926w"
          id="logradouro"
          name="logradouro"
          placeholder="logradouro"
          type="text"
          value="Rua escorpiÃ£o"
        />
      </div>
    </div>
  </form>
</div>
`;
